service/observiq-otel-collector

```diff
@@ -0,0 +1,123 @@
+#!/bin/sh
+# observIQ OTEL daemon
+# chkconfig: 2345 99 05
+# description: observIQ's distribution of the OpenTelemetry collector
+# processname: observiq-otel-collector
+# pidfile: /var/run/observiq-otel-collector.pid
+
+# Source function library.
+if [ -e /etc/init.d/functions ]; then
+  STATUS=true
+  . /etc/init.d/functions
+fi
+
+if [ -e /lib/lsb/init-functions ]; then
+  PROC=true
+  . /lib/lsb/init-functions
+fi
+
+# Pull in sysconfig settings
+[ -f /etc/sysconfig/observiq-otel-collector ] && . /etc/sysconfig/observiq-otel-collector
+
+PROGRAM=observiq-otel-collector
+LOCKFILE=/var/lock/$PROGRAM
+OIQ_OTEL_COLLECTOR_HOME=/opt/observiq-otel-collector
+OIQ_OTEL_COLLECTOR_STORAGE=/opt/observiq-otel-collector/storage
+HTTP_PROXY=
+HTTPS_PROXY=
+# set this to a user such as observiq-otel-collector, to run as that user
+# observiq-otel-collector user is created during installation
+RUNUSER=
+PIDFILE=/var/run/observiq-otel-collector.pid
+DEBUG=false
+RETVAL=0
+start() {
+  if [ -f $PIDFILE ]; then
+    PID=$(cat $PIDFILE)
+    echo " * $PROGRAM already running: $PID"
+    RETVAL=2
+  else
+    echo " * Starting $PROGRAM"
+    if [ $DEBUG ]; then
+      COMMAND="cd /opt/observiq-otel-collector/ && nohup /opt/observiq-otel-collector/observiq-otel-collector --config config.yaml 1> /dev/null 2>> $OIQ_OTEL_COLLECTOR_HOME/log/error.log"
+    else
+      COMMAND="cd /opt/observiq-otel-collector/ && nohup /opt/observiq-otel-collector/observiq-otel-collector --config config.yaml 1>> $OIQ_OTEL_COLLECTOR_HOME/log/error.log 2>&1"
+    fi
+    export OIQ_OTEL_COLLECTOR_HOME
+    export OIQ_OTEL_COLLECTOR_STORAGE
+    export HTTP_PROXY
+    export HTTPS_PROXY
+    if [ -n "$RUNUSER" ]; then
+      su -p "$RUNUSER" -c "$COMMAND" &
+    else
+      nohup "$COMMAND" &
+    fi
+    echo $! > $PIDFILE
+    # disabling shellcheck due to this being the one documented exception. See https://www.shellcheck.net/wiki/SC2320
+    # shellcheck disable=SC2320
+    RETVAL=$?
+    [ "$RETVAL" -eq 0 ] && touch $LOCKFILE
+  fi
+}
+
+stop() {
+  if [ -f $PIDFILE ]; then
+    PID=$(cat $PIDFILE);
+    printf " * Stopping %s... " "$PROGRAM"
+    kill "$PID" > /dev/null 2>&1
```

Comment:
In the systemd file we give the service 20 seconds to shut down to avoid it forever hanging. Is there any concern here of no having the collector shutdown within 20 seconds?

Reply:
init.d startup scripts don't have a concept of timeouts. Generally on init.d systems, if the service fails to stop it is up to the user to manually clean it up. One of the biggest advantages of systemd.